pennylane/ops/qubit.py

```diff
@@ -41,6 +41,23 @@ class Hadamard(Observable, Operation):
     par_domain = None
     eigvals = pauli_eigs(1)
 
+    @staticmethod
+    def matrix():
+        r"""Matrix representation of the Hadamard operator
+        Returns:
+            array[complex]: matrix representation of the Hadamard operator
+        """
+        return np.array([[1, 1], [1, -1]], dtype=complex) / np.sqrt(2)
```

Comment:
Out of curiosity, I've never used `dtype=complex` before. Does it cause any difference compared to using a NumPy dtype?
```suggestion
        return np.array([[1, 1], [1, -1]], dtype=np.complex128) / np.sqrt(2)
```

Reply:
I've mainly used `dtype=complex` but I believe it's the same as using `dtype=np.complex128`. See [here](https://docs.scipy.org/doc/numpy-1.13.0/user/basics.types.html).